config.yaml

```diff
@@ -12,6 +12,10 @@ options:
       Indicates how NGINX should communicate with the backend service. Valid
       Values: HTTP, HTTPS, GRPC, GRPCS, AJP and FCGI.
     type: string
+  enable-access-log:
+    description: >
+      Setting this to false disables access log for the ingress using nginx.ingress.kubernetes.io/enable-access-log.
+    type: boolean
```

Comment:
I think we might want to specify a default value of `true` to make it more obvious to someone reading this.

Reply:
The problem with having a default value in a boolean configuration is that the configuration value will always exist (either `True` or `False`), thereby always overriding the relational data and essentially prohibiting controlling this option via the relation. If there's no default value, then there will be three states (`True`, `False`, or `None`), allowing the relation to take control when the configuration is `None`.

How about adding this to the description?

> If this setting is not set anywhere (config or relation), it defaults to true.

Reply:
Added, thanks!

Reply:
I don't see `default: true` here, or is `true` the default value if not informed?

Reply:
I added some explanations to the description, but I didn't change the default value. The value is `null` if not set.

The challenge of having a default boolean value is that, in the `nginx-ingress-integrator` configuration, it has higher precedence over the relation. If there's a default value for a boolean configuration, then the configuration is always defined, which means the configuration will always override the relation. This means that the relation cannot control this option.

However, we can modify the logic slightly and define disabling access logs as a "dominant gene", where either relation or config being false results in false. This approach would allow the relation to "override" the configuration when the relation data is false (*).

|                            | config is true  |  config is false  |
| --------------- | -------------- | --------------- | 
| relation is true  |       true            |        false          |
| relation is false |      false*           |        false          |
| relation unset   |       true            |        false          |

What do you think? By the way, below is the current truth table:

|                            | config is true  |  config is false  | config unset |
| --------------- | -------------- | --------------- | ------------ |
| relation is true  |       true            |        false            |      true         |
| relation is false |       true            |        false            |      false        |
| relation unset    |       true            |        false            |      true         |

Reply:
Ahh, it makes sense now. I think keeping the current truth table you showed makes total sense, and it keeps backwards-compatibility. Thanks!